### PR TITLE
Add check for clusterRoleNames

### PR DESCRIPTION
**Pull request by sam-nguyen7**
currently operator is broken when the clusterRoleName we pass in the CR does not exist on the cluster. The solution is "all-or-nothing". If any clusterRoleNames do not exist on the cluster, operator will update status and exit.

### CODE REVIEW

File: pkg/controller/subjectpermission/subjectpermission_controller.go [hunk 1]
```diff
@@ -118,6 +118,20 @@ func (r *ReconcileSubjectPermission) Reconcile(request reconcile.Request) (recon
 		return reconcile.Result{}, err
 	}
 
+	// get all ClusterRoleNames that do not exist as ClusterRole
+	clusterRoleNamesNotOnCluster := populateCrClusterRoleNames(instance, clusterRoleList)
+	if len(clusterRoleNamesNotOnCluster) != 0 {
+		// update condition if any ClusterRoleName does not exist as a ClusterRole
+		instance.Status.Conditions = controllerutil.UpdateCondition(instance.Status.Conditions, "ClusterRole for ClusterPermission does not exist", clusterRoleNamesNotOnCluster, true, managedv1alpha1.SubjectPermissionStateFailed, managedv1alpha1.ClusterRoleBindingCreated)
```
Comment (fahlmant) [inline pkg/controller/subjectpermission/subjectpermission_controller.go]: So does this condition only get added once?

File: deploy/crds/managed_v1alpha1_subjectpermission_cr.yaml [hunk 2]
```diff
@@ -4,9 +4,6 @@ metadata:
   name: example-subjectpermission
 spec:
   subjectKind: Group
-  subjectName: Test
-  permissions:
```
Comment (fahlmant) [inline deploy/crds/managed_v1alpha1_subjectpermission_cr.yaml]: Why did you remove this?

File: pkg/controller/subjectpermission/subjectpermission_controller.go [hunk 3]
```diff
@@ -118,8 +118,18 @@ func (r *ReconcileSubjectPermission) Reconcile(request reconcile.Request) (recon
 		return reconcile.Result{}, err
 	}
 
+	var temp []string
+	// all clusterrolenames for testing purposes
+	for _, i := range clusterRoleList.Items {
+		temp = append(temp, i.Name)
+	}
 	// get all ClusterRoleNames that do not exist as ClusterRole
+	reqLogger.Info(fmt.Sprintf("clusterRoleName from CR: %+v", instance.Spec.ClusterPermissions))
```
Comment (fahlmant) [inline pkg/controller/subjectpermission/subjectpermission_controller.go]: Extra logs?

File: pkg/controller/subjectpermission/subjectpermission_controller.go [hunk 4]
```diff
@@ -118,8 +118,18 @@ func (r *ReconcileSubjectPermission) Reconcile(request reconcile.Request) (recon
 		return reconcile.Result{}, err
 	}
 
+	var temp []string
+	// all clusterrolenames for testing purposes
+	for _, i := range clusterRoleList.Items {
+		temp = append(temp, i.Name)
```
Comment (fahlmant) [inline pkg/controller/subjectpermission/subjectpermission_controller.go]: Testing?

File: deploy/crds/managed_v1alpha1_subjectpermission_cr.yaml (hunk 2 above)
Comment (sam-nguyen7) [reply deploy/crds/managed_v1alpha1_subjectpermission_cr.yaml]: testing with ClusterPermission in this CR specifically. I have other CRs where I test permissions, and both together

File: pkg/controller/subjectpermission/subjectpermission_controller.go (hunk 1 above)
Comment (sam-nguyen7) [reply pkg/controller/subjectpermission/subjectpermission_controller.go]: Yes, per discussion on Slack following the "all or nothing" approach. and one condition for clusterPermission and one condition for permission

File: pkg/controller/subjectpermission/subjectpermission_controller.go (hunk 4 above)
Comment (sam-nguyen7) [reply pkg/controller/subjectpermission/subjectpermission_controller.go]: yes removed

File: pkg/controller/subjectpermission/subjectpermission_controller.go (hunk 3 above)
Comment (sam-nguyen7) [reply pkg/controller/subjectpermission/subjectpermission_controller.go]: removed

File: pkg/controller/subjectpermission/subjectpermission_controller.go [hunk 5]
```diff
@@ -255,22 +283,24 @@ func newClusterRoleBinding(clusterRoleName, subjectName string, subjectKind stri
...
+	var found bool
 
-	// for every cluster role names on cluster, loop through all crClusterRoleNames, if it doesn't exist then append
-	for _, i := range onClusterItems {
-		//name := i.Name
-		for _, a := range crClusterRoleNames {
-			if i.Name != a {
-				crClusterRoleNameList = append(crClusterRoleNameList, a)
+	// for every CR clusterRoleNames, loop through all k8s lusterRoles, if it doesn't exist then append
+	for _, i := range crClusterRoleNames {
```
Comment (jharrington22) [inline pkg/controller/subjectpermission/subjectpermission_controller.go]: Could this be a function?

Comment (sam-nguyen7) [reply pkg/controller/subjectpermission/subjectpermission_controller.go]: the nested for loop is the whole logic/reason of the function, not sure if moving it to another helper function makes sense

Comment (jharrington22) [reply pkg/controller/subjectpermission/subjectpermission_controller.go]: Isn't this the same logic as https://github.com/openshift/rbac-permissions-operator/pull/32/files#diff-ef4d9a930cf76db3ac6ca4d543b6267dR25-R35

Comment (sam-nguyen7) [reply pkg/controller/subjectpermission/subjectpermission_controller.go]: It's very similar but not the same @jharrington22

Comment (sam-nguyen7) [reply pkg/controller/subjectpermission/subjectpermission_controller.go]: this one loops through []string, the other one loops through []Permissions

Comment (sam-nguyen7) [reply pkg/controller/subjectpermission/subjectpermission_controller.go]: future feature: make this a type agnostic way to do the nested for loop
